### PR TITLE
Adding framework for production checks

### PR DESCRIPTION
**Pull request by lasryaric**
Addressing: 

https://github.com/dust-tt/tasks/issues/234

We won't start the workflow until we actually have a real check running.

The demo check (super_check) can be removed from this PR but it's an example of how to write the first one until we have one in the codebase.

### CODE REVIEW

File: front/production_checks/lib/libcheck.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export type CheckFunction = (
```
Comment (spolu) [inline front/production_checks/lib/libcheck.ts:1]: no need to repeat lib?

File: front/production_checks/lib/libcheck.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  checkName: string,
+  reportSuccess: (reportPayload: unknown) => void,
+  reportFailure: (reportPayload: unknown, message: string) => void
+) => Promise<void>;
+
+export type Check = {
+  name: string;
```
Comment (spolu) [inline front/production_checks/lib/libcheck.ts:8]: intervalHours some we may not want to run hourly?

File: front/production_checks/temporal/workflows.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { proxyActivities } from "@temporalio/workflow";
+
+import type * as activities from "@app/production_checks/temporal/activities";
+
+const { runAllChecksActivity } = proxyActivities<typeof activities>({
+  startToCloseTimeout: "1 hour",
+  heartbeatTimeout: "5 minutes",
```
Comment (spolu) [inline front/production_checks/temporal/workflows.ts:7]: Are we heartbeating in between checks? this will break fast otherwise?

Comment (lasryaric) [reply front/production_checks/temporal/workflows.ts:7]: yes correct.

**Review by spolu — DISMISSED** [review]
LGTM

Let's merge and do a clean PR with a first check as example for the whole team

File: front/production_checks/lib/libcheck.ts (hunk 2 above)
Comment (lasryaric) [reply front/production_checks/lib/libcheck.ts]: Mentionned in the specs that we will do that in a follow up version of this framework. Just to get it quickly out of the door.

File: front/production_checks/lib/libcheck.ts (hunk 1 above)
Comment (lasryaric) [reply front/production_checks/lib/libcheck.ts]: I renamed it from `/lib/libcheck.ts` to `types/check.ts`

**Review by lasryaric — COMMENTED** [review]
Cleaned up a few things you suggested.

I will merge and send a PR for a real check soon.

**Pull request merged**